ORStools/utils/gui.py

```diff
@@ -0,0 +1,275 @@
+        else:
+            color_string = 'rgba(0,0,0,.3)'
+        res = re.sub(r'(<a href.*?)>',
+                     r'\1 style="color: {};">'.format(color_string),
+                     html)
+        if wrap_color:
+            res = '<span style="color: {};">{}</span>'.format(color_string,
+                                                              res)
+        return res
+
+    @staticmethod
+    def get_icon(icon: str) -> QIcon:
+        """
+        Returns a plugin icon
+        :param icon: icon name (svg file name)
+        :return: QIcon
+        """
+        path = GuiUtils.get_icon_svg(icon)
+        if not path:
+            return QIcon()
+
+        return QIcon(path)
+
+    @staticmethod
+    def get_icon_svg(icon: str) -> str:
+        """
+        Returns a plugin icon's SVG file path
+        :param icon: icon name (svg file name)
+        :return: icon svg path
+        """
+        path = os.path.join(
+            os.path.dirname(__file__),
+            '..',
+            'gui/img',
+            icon)
+        logger.log(path)
+        if not os.path.exists(path):
+            return ''
+
+        return path
+
+    @staticmethod
+    def get_icon_pixmap(icon: str) -> QPixmap:
+        """
+        Returns a plugin icon's PNG file path
+        :param icon: icon name (png file name)
+        :return: icon png path
+        """
+        path = os.path.join(
+            os.path.dirname(__file__),
+            '..',
+            'icons',
+            icon)
+        if not os.path.exists(path):
+            return QPixmap()
+
+        im = QImage(path)
+        return QPixmap.fromImage(im)
+
+    @staticmethod
+    def get_svg_as_image(icon: str, width: int, height: int,
+                         background_color: Optional[QColor] = None,
+                         device_pixel_ratio: float = 1) -> QImage:
+        """
+        Returns an SVG returned as an image
+        """
+        path = GuiUtils.get_icon_svg(icon)
+        if not os.path.exists(path):
+            return QImage()
+
+        renderer = QSvgRenderer(path)
+        image = QImage(int(width * device_pixel_ratio),
+                       int(height * device_pixel_ratio),
+                       QImage.Format_ARGB32)
+        image.setDevicePixelRatio(device_pixel_ratio)
+        if not background_color:
+            image.fill(Qt.transparent)
+        else:
+            image.fill(background_color)
+
+        painter = QPainter(image)
+        painter.scale(1 / device_pixel_ratio,
+                      1 / device_pixel_ratio)
+        renderer.render(painter)
+        painter.end()
+
+        return image
+
+    @staticmethod
+    def get_ui_file_path(file: str) -> str:
+        """
+        Returns a UI file's path
+        :param file: file name (uifile name)
+        :return: ui file path
+        """
+        path = os.path.join(
+            os.path.dirname(__file__),
+            '..',
+            'gui',
+            file)
```

Comment:
This is highly dependent on the code layout of the plugin - in our case, it works, but it feels way more generic than it is?